### ISSUE TITLE
Does amgcl support the trillinos csrmatrix format.

### ISSUE THREAD

**Issue by ztdepztdep** [ddemidov/amgcl#225]
I have a code writing with trillinos, and i want to call the amgcl routine from my trillinos code. Does amgcl supoor thtis?

**Comment by ddemidov** [comment]
There is the [`epetra_map`](https://github.com/ddemidov/amgcl/blob/master/amgcl/adapter/epetra.hpp) adapter that should allow to use Trilinos matrices as inputs to amgcl solvers.

**Comment by ztdepztdep** [comment]
Could you please give me an example about how to use it.

**Comment by ddemidov** [comment]
I don't have much experience with Trilinos, may be @RiccardoRossi may comment?

Also, make sure you have enabled full optimization for compilation of the amgcl test.

**Comment by RiccardoRossi** [comment]
mmm,

as i understand you have a backend in the middle, and as i understand that is making a copy of the matrix. @ddemidov am I wrong?

having said this we benchmarked against ML and it was very competitive... are u sure you properly defined the NDEBUG flag?

**Comment by ddemidov** [comment]
Yes, there is a copy being made from the adapter into amgcl distributed matrix. @ztdepztdep, if you measure the construction and the application of the solver separately, it should be clear which part takes more time (the matrix is copied inside the constructor).

**Comment by ztdepztdep** [comment]
> 

Could you please give me some advices about how to define NDEBUG in the cmake file?

**Comment by ddemidov** [comment]
Cmake should do this for you if you set CMAKE_BUILD_TYPE=Release
